File: ThreeDojo/WebContent/js/my/DependencyPropertyHelper.js
define("",["dojo/_base/declare"], function(declare){
	var DependencyPropertyHelper = {
		//public static ValueSource GetValueSource(DependencyObject dependencyObject, DependencyProperty dependencyProperty)
		GetValueSource:function(dependencyObject,dependencyProperty) { 
	        if (dependencyObject == null)
	            throw new Error("dependencyObject");
	        if (dependencyProperty == null)
	            throw new Error("dependencyProperty"); 
	
	        dependencyObject.VerifyAccess(); 
	
	        /*boolean*/ var hasModifiers, isExpression, isAnimated, isCoerced, isCurrent;
	        /*BaseValueSourceInternal*/var source = dependencyObject.GetValueSource(dependencyProperty, null, hasModifiers, isExpression, isAnimated, isCoerced, isCurrent); 
	
	        return new ValueSource(source, isExpression, isAnimated, isCoerced, isCurrent);
	    }
	};
	return DependencyPropertyHelper;	
}); 

File: org.summer.view.widget/src/org/summer/view/widget/DependencyPropertyHelper.java
package org.summer.view.widget;


/// <summary>
/// Helper class for miscellaneous framework-level features related 
/// to DependencyProperties.
/// </summary>
public class DependencyPropertyHelper
{ 
    /// <summary>
    /// Return the source of the value for the given property. 
    /// </summary> 
    public static ValueSource GetValueSource(DependencyObject dependencyObject, DependencyProperty dependencyProperty)
    { 
        if (dependencyObject == null)
            throw new ArgumentNullException("dependencyObject");
        if (dependencyProperty == null)
            throw new ArgumentNullException("dependencyProperty"); 

        dependencyObject.VerifyAccess(); 

        boolean hasModifiers, isExpression, isAnimated, isCoerced, isCurrent;
        BaseValueSourceInternal source = dependencyObject.GetValueSource(dependencyProperty, null, /*out*/ hasModifiers, /*out*/ isExpression, /*out*/ isAnimated, /*out*/ isCoerced, /*out*/ isCurrent); 

        return new ValueSource(source, isExpression, isAnimated, isCoerced, isCurrent);
    }
} 